src/program.ts

```diff
@@ -4533,6 +4534,26 @@ export class Class extends TypedElement {
     }
     return out;
   }
+
+  getVisitPrototype(compiler: Compiler): DeclaredElement | null {
+    const visitPrototype = this.getMember("__visit");
+    if (visitPrototype == null) {
+      return null;
+    }
+    if (this.isDeclaredInLibrary) {
+      return visitPrototype;
+    }
+    const decorator = visitPrototype.declaration.decorators;
+    if (!decorator || (decorator && decorator.every((v) => v.decoratorKind != DecoratorKind.UNSAFE))) {
+      compiler.warning(DiagnosticCode.override___visit_need_unsafe_decorator,visitPrototype.declaration.name.range);
```

Comment:
And I think it should be an error, not warning. @dcodeIO wdyt?

Reply:
If the compiler ignores it when the decorator is not present, I think a warning is sufficient so one could theoretically use the name `__visit` otherwise. Seems unlikely, but probably does not harm to be a little more conservative about it.

Reply:
I don't think the user will want to use `__visit` specifically. It's more likely that it's the visiting GC roots that he wants to owerride. But he may not notice the warning, especially if there are a lot of such warnings, but concerning some other unimportant things. That's why I think it's better to make it an error